### PR TITLE
feat: add feedback for sort and filter button

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Feature**

This PR resolves [FX-2927]

### Description
When the user clicks on the "sort & filter" button, there should be a visual feedback on the click. Jonathan [suggested](https://artsyproduct.atlassian.net/browse/FX-2927?focusedCommentId=37957) that when you click on the button, change the color to `blue100`

#### Demo

https://user-images.githubusercontent.com/3513494/121333140-3417a280-c921-11eb-815d-f1ba8df0fea7.mp4

https://user-images.githubusercontent.com/3513494/121335895-aee1bd00-c923-11eb-8445-cfaa49c8cd1b.mp4

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[FX-2927]: https://artsyproduct.atlassian.net/browse/FX-2927

### CODE REVIEW

File: src/lib/Scenes/Collection/Components/CollectionArtworksFilter.tsx [hunk 1]
```diff
@@ -150,14 +150,18 @@ export const CollectionArtworksFilter: React.FC<FilterProps> = ({ collection, an
...
+                onPress={openFilterArtworksModal}
+                render={({ color }) => (
+                  <Flex flexDirection="row" alignItems="center">
+                    <FilterIcon fill={color} width="20px" height="20px" />
+                    <Text variant="subtitle" color={color}>
+                      Sort & Filter
+                    </Text>
+                  </Flex>
+                )}
+              />
```
Comment (mdole) [inline src/lib/Scenes/Collection/Components/CollectionArtworksFilter.tsx:164]: @dimatretyak could this be accomplished using React Native's `<TouchableHighlight>` component, or would the icon and text not correctly change color?

**Review by mdole — COMMENTED** [review]
If we can't use `TouchableHighlight`, then this looks good to me! Just wanted to confirm that we weren't building a new component unnecessarily 🙂

File: src/lib/Scenes/Collection/Components/CollectionArtworksFilter.tsx (hunk 1 above)
Comment (dimatretyak) [reply src/lib/Scenes/Collection/Components/CollectionArtworksFilter.tsx:164]: @mdole Yes, you're right. Using `TouchableHighlight` we can't set the selected color for the text and icon using 

https://user-images.githubusercontent.com/3513494/121370248-3c81d480-c945-11eb-905c-db55970e9f0d.mp4

Comment (mdole) [reply src/lib/Scenes/Collection/Components/CollectionArtworksFilter.tsx:164]: Got it, thanks for confirming!

**Review by mdole — APPROVED** [review]
@dimatretyak looks like we have a merge conflict, but otherwise this seems ready to me!

**Comment by dimatretyak** [comment]
> @dimatretyak looks like we have a merge conflict, but otherwise this seems ready to me!

Done

**Pull request merged**